source/includes/monitoring_JMX.kt

```diff
@@ -0,0 +1,30 @@
+import com.mongodb.kotlin.client.MongoClient
+import org.bson.json.JsonWriterSettings
+import com.mongodb.event.*
+import com.mongodb.MongoClientSettings
+import com.mongodb.ConnectionString
+import com.mongodb.management.JMXConnectionPoolListener
+
+
+fun main() {
+    val uri = "<connection string uri>"
+
+// Instantiate your JMX listener
+    val connectionPoolListener = JMXConnectionPoolListener()
+
+// Include the listener in your client settings
+    val settings = MongoClientSettings.builder()
+        .applyConnectionString(ConnectionString(uri))
+        .addCommandListener(connectionPoolListener)
```

Comment:
We're not adding a command listener, the goal is to add a connection pool?
```suggestion
        .applyToConnectionPoolSettings { it.addConnectionPoolListener(connectionPoolListener) }
```